### ISSUE TITLE
Asset optimization (Image tag) doesn't work in dev server, but works after build

### ISSUE THREAD

**Issue by Hubro** [withastro/astro#8486]
### Astro Info

```block
npx astro info
Astro                    v3.0.12
Node                     v19.7.0
System                   Linux (x64)
Package Manager          npm
Output                   static
Adapter                  none
Integrations             none
```

### Describe the Bug

The `Image` component works after `npm run build` but doesn't work in dev server.

Starting with the default Astro template (`npm create astro@latest`) and adding:

```diff
--- a/src/pages/index.astro
+++ b/src/pages/index.astro
@@ -1,5 +1,7 @@
 ---
+import { Image } from "astro:assets";
 
+import catPicture from "../assets/cat.jpg";
 ---
 
 <html lang="en">
@@ -11,6 +13,8 @@
                <title>Astro</title>
        </head>
        <body>
-               <h1>Astro</h1>
+               <h1>Optimized cat picture:</h1>
+
+               <Image alt="cat" src={catPicture} />
        </body>
 </html>
```

And running `npm run dev`, I get:

![image](https://github.com/withastro/astro/assets/597206/71dd3049-43e4-4d1c-bdce-8557477134a1)

(See example here: https://github.com/Hubro/astro-image-optimization-test)

However, when running `npm run build`, the image gets properly optimized and linked:

```
$ tree dist
dist
├── _astro
│   ├── cat.f57e8aae.jpg
│   └── cat.f57e8aae_Z13uAgK.webp
├── favicon.svg
└── index.html
```

![image](https://github.com/withastro/astro/assets/597206/92f7d25d-d5ab-49e5-98f8-91136bd6d306)

---

The request on the dev server looks like (copied from dev console):

```
Request URL: http://localhost:4321/_image?href=%2F%40fs%2Fhome%2Ftomas%2FDesktop%2Fimage-optimization-test%2Fsrc%2Fassets%2Fcat.jpg%3ForigWidth%3D1280%26origHeight%3D874%26origFormat%3Djpg&f=webp
Request Method: GET
Status Code: 404 Not Found
```

I'm extremely confused that this doesn't work, considering I'm using the default template and following [the documentation](https://docs.astro.build/en/guides/images/#image--astroassets) verbatim. I can't possibly be the first person to run into this? :thinking: 

### What's the expected result?

Expected the image to also work in dev

### Link to Minimal Reproducible Example

https://github.com/Hubro/astro-image-optimization-test

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by Hubro** [comment]
It might be related to the first one. I noticed earlier that `localhost` on my system resolves `::1`, which I believe is default on Arch linux (or at least it was default when I installed this system).

I edited my `/etc/hosts` file to make sure `localhost` only resolves to `127.0.0.1` and restarted the dev server, and that resolved the issue.

Should I close this, or would you agree this is a bug?

**Comment by Princesseuh** [comment]
I'll close this one since the first issue linked also cover this, thank you nonetheless for the report!
